### PR TITLE
Feat/m1 manual build

### PR DESCRIPTION
**Pull request by McPatate**
#932

### CODE REVIEW

File: bindings/python/tokenizers_manual_m1_build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+
+MACOSX_DEPLOYMENT_TARGET=10.11 SDKROOT="/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" CFLAGS="-I/usr/include/openssl -I/usr/local/opt/readline/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include" CPPFLAGS="-I/usr/local/opt/zlib/include" LDFLAGS="-L/usr/lib -L/usr/local/opt/readline/lib" pyenv install <wtv_version>
+
+~/.pyenv/versions/<wtv_version>/bin/pip install setuptools wheel setuptools-rust==0.11.3 --ignore-installed --force-reinstall
+
+MACOSX_DEPLOYMENT_TARGET=10.11 ~/.pyenv/versions/<wtv_version>/bin/python setup.py bdist_wheel
+
```
Comment (Narsil) [inline bindings/python/tokenizers_manual_m1_build.sh:8]:
```suggestion
for VARIABLE in "3.7.12" "3.8.12" "3.9.10" "3.10.2"
do
    MACOSX_DEPLOYMENT_TARGET=10.11 SDKROOT="/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk" CFLAGS="-I/usr/include/openssl -I/usr/local/opt/readline/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include" CPPFLAGS="-I/usr/local/opt/zlib/include" LDFLAGS="-L/usr/lib -L/usr/local/opt/readline/lib" pyenv install $VARIABLE

    ~/.pyenv/versions/$VARIABLE/bin/pip install setuptools wheel setuptools-rust==0.11.3 --ignore-installed --force-reinstall

    MACOSX_DEPLOYMENT_TARGET=10.11 ~/.pyenv/versions/$VARIABLE/bin/python setup.py bdist_wheel
done
```

Is that OK ? (Fix the library versions if you want)

File: bindings/python/.cargo/config.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[target.x86_64-apple-darwin]
+rustflags = [
+  "-C", "link-arg=-undefined",
+  "-C", "link-arg=dynamic_lookup",
+  "-C", "link-arg=-mmacosx-version-min=10.11",
```
Comment (Narsil) [inline bindings/python/.cargo/config.toml:5]: Do we still have a link where this is explained ? I can only find the `MACOSX_DEPLOYMENT_TARGET` with my google fu.

(It's not super important)

**Review by Narsil — COMMENTED** [review]
Thank you very much !

**Review by Narsil — APPROVED** [review]

File: bindings/python/.cargo/config.toml (hunk 2 above)
Comment (McPatate) [reply bindings/python/.cargo/config.toml:5]: It comes from this thread :
https://users.rust-lang.org/t/compile-rust-binary-for-older-versions-of-mac-osx/38695

Comment (McPatate) [reply bindings/python/.cargo/config.toml:5]: I think we should leave that here for now. I'll test on next release without that flag to see if it still works.

**Pull request merged**